app/src/org/commcare/android/crypt/EncryptionIO.java

```diff
@@ -0,0 +1,95 @@
+package org.commcare.android.crypt;
+
+import android.util.Log;
+
+import org.commcare.android.javarosa.AndroidLogger;
+import org.javarosa.core.services.Logger;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Methods for dealing with encrypted input/output.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class EncryptionIO {
+    private final static String TAG = EncryptionIO.class.getSimpleName();
+
+    public static OutputStream createFileOutputStream(String filename,
+                                                      SecretKeySpec symetricKey)
+            throws FileNotFoundException {
+        final File path = new File(filename);
+        FileOutputStream fos = new FileOutputStream(path);
+        if (symetricKey == null) {
+            return fos;
+        } else {
+            try {
+                Cipher cipher = Cipher.getInstance("AES");
+                cipher.init(Cipher.ENCRYPT_MODE, symetricKey);
+                return new CipherOutputStream(fos, cipher);
```

Comment:
Writing to a buffered stream should still help with file writes. The data is in memory but we're still writing it in a way that is chunked out of convenience by whatever structure the incoming stream decides (IE: Regardless of whether it's 2 bytes or 134 bytes it'll still try to dump them whenever it has them). Using a buffered output steam lets the writer write in whatever way is convenient without actually starting the file I/O until it can write a nice clean chunk, which should speed things up on both ends.

Basically if one end of a stream is I/O backed that end of the stream should almost always be buffered unless we know for sure that it already is.